extensions/wikia/SoundCloudTag/SoundCloudTagController.class.php

```diff
@@ -0,0 +1,44 @@
+<?php
+class SoundCloudTagController extends WikiaParserTagController {
+	const PARSER_TAG_NAME = 'soundcloud';
+	const API_ENDPOINT = 'https://w.soundcloud.com/player/?';
+
+
+	public static function onParserFirstCallInit( Parser $parser ) {
+		$parser->setHook( self::PARSER_TAG_NAME, [ new static(), 'renderTag' ] );
+		return true;
+	}
+
+	public function renderTag( $input, array $args, Parser $parser, PPFrame $frame ) {
+		return Html::element(
+			'iframe',
+			$this->prepareAttributes( $args ),
+			wfMessage( 'soundcloud-tag-could-not-render' )->text()
+		);
+	}
+
+	private function prepareAttributes( array $args ) {
+		$allowedArgs = ['width', 'height', 'scrolling', 'frameborder', 'src'];
+		$attributes = [];
+
+		if ( array_key_exists('style', $args) ) {
+			$attributes['style'] = Sanitizer::checkCss( $args['style'] );
+		}
+
+		foreach ( $allowedArgs as $name ) {
+			if ( isset( $args[$name] ) ) {
+				$attributes[$name] = $args[$name];
+			}
+		}
+
+		$attributes['data-tag'] = 'soundcloud';
+		$attributes['sandbox'] = 'allow-scripts allow-same-origin';
```

Comment:
Why would we allow users to define things like that? What's the business value? I found options that are far more useful and IMO they should be here instead of technicalities:
- `auto_play`
- `color`
- `hide_related`
- `show_comments`
- `show_user`
- `show_reposts`
- `visual`